File: code/c-sharp/src/OopSolidSample/Controllers/BlackjackController.cs
using System.Web.Mvc;
using MvcContrib.Attributes;
using OopSolidSample.Domain;
using OopSolidSample.Models.Blackjack;

namespace OopSolidSample.Controllers
{
    public class BlackjackController : Controller
    {
        [AcceptGet]
        public ActionResult Index()
        {
            var model = new BlackjackDrawViewModel();

            return View();
        }

        [AcceptPost]
        public ActionResult Draw()
        {
            var model = new BlackjackDrawViewModel();
            var game = new BlackjackGame();
            var drawResult = game.Draw();

            foreach (var card in drawResult.Cards)
            {
                if (!string.IsNullOrEmpty(model.Cards))
                    model.Cards += ", ";
                model.Cards += card.Description;
            }

            model.Score = drawResult.Score;
            return View(model);
        }
    }
}

File: code/ruby/app/controllers/blackjack_controller.rb
class BlackjackController < ApplicationController
  def index
    
  end

  def draw
    game = BlackjackGame.new
    draw_result = game.draw

    @cards = ''
    draw_result.cards.each {|card|
      if @cards.length > 0
        @cards += ', '
      end
      @cards += card.description
    }

    @score = draw_result.score
  end
end
